application/controllers/admin/useraction.php

```diff
@@ -643,28 +643,31 @@ public function personalsettings()
                     $error = $oUserModel->checkPasswordStrength($newPassword);
                     if ($error) {
                         Yii::app()->setFlashMessage(gT($error), 'error');
-                        $this->getController()->redirect(array("admin/user/sa/personalsettings"));
+                        return $this->getController()->redirect(array("admin/user/sa/personalsettings"));
                     }
-                } elseif (!$oUserModel->checkPassword($oldPassword)) {
+                }
+                if (!$oUserModel->checkPassword($oldPassword)) {
                     // Always check password
                     Yii::app()->setFlashMessage(gT("Your new password was not saved because the old password was wrong."), 'error');
-                    $this->getController()->redirect(array("admin/user/sa/personalsettings"));
-                } elseif (trim($oldPassword) === trim($newPassword)) {
+                    return $this->getController()->redirect(array("admin/user/sa/personalsettings"));
+                }
+                if (trim($oldPassword) === trim($newPassword)) {
                     //First test if old and new password are identical => no need to save it (or ?)
                     Yii::app()->setFlashMessage(gT("Your new password was not saved because it matches the old password."), 'error');
-                    $this->getController()->redirect(array("admin/user/sa/personalsettings"));
-                } elseif (trim($newPassword) !== trim($repeatPassword)) {
+                    return $this->getController()->redirect(array("admin/user/sa/personalsettings"));
+                }
+                if (trim($newPassword) !== trim($repeatPassword)) {
                     //Then test the new password and the repeat password for identity
                     Yii::app()->setFlashMessage(gT("Your new password was not saved because the passwords did not match."), 'error');
-                    $this->getController()->redirect(array("admin/user/sa/personalsettings"));
+                    return $this->getController()->redirect(array("admin/user/sa/personalsettings"));
                 //Now check if the old password matches the old password saved
-                } elseif (empty(trim($newPassword))) {
+                }
+                if (empty(trim($newPassword))) {
                     Yii::app()->setFlashMessage(gT("The password can't be empty."), 'error');
-                    $this->getController()->redirect(array("admin/user/sa/personalsettings"));
-                } else {
-                    // We can update
-                    $oUserModel->setPassword($newPassword);
+                    return $this->getController()->redirect(array("admin/user/sa/personalsettings"));
```

Comment:
redirect is a redirect, no return please

Reply:
Here is needed so action returns immediately showing the appropiate flash message (https://www.yiiframework.com/doc/api/2.0/yii-web-controller#redirect()-detail).
Without a return, it would show "Your personal settings were successfully saved." leading to confusion.
Otherwise it wold require more changes on then follow up code to check if errors happened on password change for showing the appropiate flash message.
Any suggestion at this respect?

Reply:
I mean : 

Since `$this->getController()->redirect(array("admin/user/sa/personalsettings"));` is a redirect : you CAN'T return it …

It's an end : no return 

See https://www.yiiframework.com/doc/api/1.1/CHttpRequest#redirect-detail

return void …

Reply:
OK, thanks, you are right!
I'll change that.